### PR TITLE
What's New - Remove hardcoded values from service calls

### LINKED ISSUE #14584: Add a service to retrieve "What's New" data from the backend
The What's New / Feature Announcement modal will be populated with data coming from the backend.
For this, there a remote service has been added to `WordPressKit` (https://github.com/wordpress-mobile/WordPressKit-iOS/pull/276).

### PR DESCRIPTION
**Pull request by Gio2018**
Fixes #14584

This removes hardcoded values and passes the actual app version to the endpoint.

To test:

- Go to Me/App Settings/What's New in WordPress
- Make sure that there are three announcement shown, with the same content (title, subtitle and image) as the screenshots below
- The "Find out more" button, at the moment, should open WordPress.com in the default browser (this will change in the future)
- The "Continue" button should dismiss the screen
- Try different devices/iOS versions and orientations and make sure that the views look right

Note: the content of the announcements is purely for testing, and does not reflect actual features of WordPress-iOS

iPhone | iPad
--- | ---
 | 

PR submission checklist:

- [x] ~~I have considered adding unit tests where possible.~~ Future PR
- [x] ~~I have considered adding accessibility improvements for my changes.~~ Future PR
- [x] ~~I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.~~ Not released.

### CODE REVIEW

File: Podfile.lock [hunk 1]
```diff
@@ -741,7 +741,7 @@ SPEC CHECKSUMS:
   React-RCTVibration: 20deb1f6f001000d1f2603722ec110c66c74796b
   ReactCommon: 48926fc48fcd7c8a629860049ffba9c23b4005dc
   ReactNativeDarkMode: f61376360c5d983907e5c316e8e1c853a8c2f348
-  RNCMaskedView: 5a8ec07677aa885546a0d98da336457e2bea557f
```
Comment (Gio2018) [inline Podfile.lock:744]: !@#$%??&$

Comment (mindgraffiti) [reply Podfile.lock:744]: 🧸

File: WordPress/Classes/ViewRelated/What's New/Data store/AnnouncementsStore.swift [hunk 2]
```diff
@@ -71,3 +71,14 @@ class RemoteAnnouncementsStore: AnnouncementsStore {
                                               localeKey: WordPressComRestApi.LocaleKeyV2)
     }
 }
+
+
+private extension RemoteAnnouncementsStore {
+    enum Identifiers {
+        // 2 is the identifier of WordPress-iOS in the backend
+        static let appId = "2"
```
Comment (mindgraffiti) [inline WordPress/Classes/ViewRelated/What's New/Data store/AnnouncementsStore.swift:79]: +1

**Review by mindgraffiti — APPROVED** [review]
✅  Code
✅  UI tested

**Pull request merged**